File: SqlSanitizer.Api/Models/FormatRequest.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace SqlSanitizer.Api.Models
{
    public enum Casing
    {
        Default,
        Upper,
        Lower,
        Capitalize
    }
    
    public class FormatRequest
    {
        public string SqlQuery { get; set; }
        public string[] CharsToRemove { get; set; }
        public bool Reindent { get; set; }
        public int IndentWidth { get; set; }
        [JsonConverter(typeof(StringEnumConverter))]
        public Casing IdentifierCase { get; set; }
        [JsonConverter(typeof(StringEnumConverter))]
        public Casing KeywordCase { get; set; }
        public bool StripComments { get; set; }
        public  SqlParameter[] Parameter { get; set; }
    }
}

File: ion-sanitizer/src/app/models/format-request.ts
import {SqlParameter} from './sql-parameter';

export interface FormatRequest {
    charsToRemove: string[];
    reindent: boolean;
    indentWidth: number;
    identifierCase: string;
    keywordCase: string;
    stripComments: boolean;
    sqlQuery: string;
    parameter: SqlParameter[];
}
